error.go

```diff
@@ -0,0 +1,21 @@
+package oidc
+
+import "errors"
+
+var (
+	ErrIssuerInvalid = errors.New("issuer does not match")
```

Comment:
How will we decide if an error shall be defined here or if we just return an error using fmt.Errorf()?

Reply:
As we work on the layering, whatever errors are returned from the inner layers should have typed declarations such that they can be checked and reacted to. 

The outer layers (the "flows" currently) do not return errors and can simply return an error string for printing.

I was thinking that around the "oauth client" code it might make sense to be a bit more formal with the contract by having typed errors.